source/slang/slang-parser.cpp

```diff
@@ -3882,6 +3882,14 @@ namespace Slang
         return decl;
     }
 
+    static NodeBase* parseIfLetDecl(
```

Comment:
The reason I create a new function is because in `parseLetDecl`, there is a check for semicolon
```
    static void parseModernVarDeclCommon(
        Parser*         parser,
        VarDecl* decl)
    {
        parseModernVarDeclBaseCommon(parser, decl);
        expect(parser, TokenType::Semicolon);  -> here
    }

```
but there won't be `;` in the `if(let ...)` case.

Reply:
Let's not make that an expr because it shouldn't be treated as one. Just write all the parsing logic in the parseIfLetStmt function.

Reply:
Sounds like we have to create a `parseIfLetStmt` function, so we do want to peek next couple tokens until we find a `let` keyword?